packages/generator/src/edmx-to-vdm/common/entity.ts

```diff
@@ -130,19 +131,26 @@ const propertyFieldType = (type: string): string | undefined =>
 
 export function joinEntityMetadata<
   EntitySetT extends EdmxEntitySetBase,
-  EntityTypeT extends EdmxEntityTypeBase<any>
+  EntityTypeT extends EdmxEntityTypeBaseNamespaced<any>
 >(
   entitySets: EntitySetT[],
   entityTypes: EntityTypeT[],
-  namespace: string,
+  // TODO 1584
+  namespace: string[],
   swagger?: SwaggerMetadata
 ): JoinedEntityMetadata<EntitySetT, EntityTypeT>[] {
   return entitySets.map(entitySet => {
-    // We assume metadata files to have a maximum of two schemas currently
-    // So entitySet.EntityType.split('.').slice(-1)[0] that we will only find one matching entry (and thus never forget anything)
-    const entityType = entityTypes.find(
-      t => t.Name === entitySet.EntityType.split('.').slice(-1)[0]
+    let entityType = entityTypes.find(
+      t => `${t.Namespace}.${t.Name}` === entitySet.EntityType
```

Comment:
can you explain give an example under which conditions the `entityType` us undefined and the second `split` on the set finds something.

Reply:
Below is an example of multi-schema, where the entitySet and entityType are linked. After this PR, our generator should be able to handle all the valid edmx that provides proper namespace information (`Namespace1` in the example). 
However, this PR will also apply stricter rules for the edmx file, which means if the namespace information is not proper, we cannot match the `Namespace1.A_TestEntityType1`.
I don't want to introduce breaking changes, if some S/4 edmx files have such issues, providing valid edmx is not our responsibility though. 
I created a ticket for handling/discussing this. 
For the time being, keeping the backwards solution that use the old logic can save us from this situation.

```
<Schema Namespace="Namespace1" xmlns="http://docs.oasis-open.org/odata/ns/edm">
      <EntityType Name="A_TestEntityType1" sap:content-version="1">
...
<Schema Namespace="Namespace5" xmlns="http://docs.oasis-open.org/odata/ns/edm">
      <EntityContainer Name="Container">
        <EntitySet Name="A_TestEntity1" EntityType="Namespace1.A_TestEntityType1" />
...
```